pkg/kubelet/volumes.go

```diff
@@ -87,11 +91,22 @@ func (vh *volumeHost) NewWrapperUnmounter(volName string, spec volume.Spec, podU
 		// Not found but not an error
 		return nil, nil
 	}
+
 	c, err := plugin.NewUnmounter(spec.Name(), podUID)
-	if err == nil && c == nil {
-		return nil, errUnsupportedVolumeType
+	if err != nil {
+		return nil, err
+	}
+	if c != nil {
+		// return good volume unmounter
+		return c, nil
 	}
-	return c, nil
+
+	// generic volume type error
+	return nil, createVolumeTypeError(volName)
+}
+
+func createVolumeTypeError(volName string) error {
+	return fmt.Errorf("Unable to mount volume or unsupported volume type for volume: %s for plugin: Unknown - check logs or pod describe events for possible root cause", volName)
```

Comment:
Hmm... I am not sure if we should prefix errors with all of that.
For example I think that:

`endpoints "glusterfs-cluster" not found`

provides better use experience than:

`FailedMount Unable to mount volumes for pod "bb-gluster-pod2_default(9fd3f3a2-f61a-11e5-b038-52540092b5fb)": failed to instantiate volume: glustervol for plugin: kubernetes.io/persistent-claim with root cause: failed to instantiate volume: pv-gluster for plugin: kubernetes.io/glusterfs with root cause: endpoints "glusterfs-cluster" not found`

Then again the underlying error might require explanation. Here I think we can get away with not prefixing and just returning the error. If the underlying error is too cryptic we can fix it at the point where it is generated.